tree/tree/src/TBasket.cxx

```diff
@@ -724,13 +724,56 @@ void TBasket::AdoptBuffer(TBuffer *user_buffer)
    fBufferRef = user_buffer;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Reset the read basket TBuffer memory allocation if needed.
+///
+/// This allows to reduce the number of memory allocation while avoiding to 
+/// always use the maximum size.
+
+void TBasket::ReadResetBuffer(Int_t basketnumber)
+{
+    // Downsize the buffer if needed.
+
+   const auto maxbaskets = fBranch->GetMaxBaskets();
+   if (!fBufferRef || basketnumber >= maxbaskets)
+      return;
+
+   Int_t curSize = fBufferRef->BufferSize();
+
+   const Float_t target_mem_ratio = fBranch->GetTree()->GetTargetMemoryRatio();
+   const auto basketbytes = fBranch->GetBasketBytes();
+
+   Int_t max_size = basketbytes[basketnumber];
+   for(Int_t b = basketnumber + 1; (b < maxbaskets) && (b < (basketnumber+10)); ++b) {
+      max_size = std::max(max_size, basketbytes[b]);
+   }
+
+   Float_t cx = 1;
+   if (fBranch->GetZipBytes())
+      cx = (fBranch->GetTotBytes()+0.00001)/fBranch->GetZipBytes();
```

Comment:
I guess that should prevent `cx` from being zero? But we never divide by `cx`, so I'm not sure why this is important.

Reply:
This is a copy paste from TBranch::Print introduced in commit cbfabd2c353 but I think this might have been for visual reasons (sorta round up).